src/sections/CustomButton.tsx

```diff
@@ -0,0 +1,54 @@
+import React from "react";
+
+interface CustomButtonProps {
+  onClick?: () => void;
+  label: string;
+  icon?: string;
+  iconPosition?: "left" | "right";
+  color?: string;
+  disabled?: boolean;
+  borderColor?: string;
+  href?: string; // Optional URL to navigate to
+  target?: string; // Optional target (_blank, _self, etc.)
+}
+
+const CustomButton: React.FC<CustomButtonProps> = ({
+  onClick,
+  label,
+  icon,
+  iconPosition = "left",
+  color = "#1C1C1C",
+  disabled = false,
+  borderColor = "#3B3B3B",
+  href,
+  target = "_self",
+}) => {
+  // Handle both onClick and href navigation
+  const handleClick = () => {
+    if (href) {
+      window.open(href, target);
+    } else if (onClick) {
+      onClick();
+    }
```

Comment:
try to use early returns

```
if(href) {
    return window.open(href, target);
}

onClick();
```

In this way, you can avoid all the messy if/else loops as well as the code looks much cleaner and easy to understand